vscode-dotnet-runtime-library/src/Acquisition/ExistingPathResolver.ts

```diff
@@ -36,21 +66,121 @@ export class ExistingPathResolver
+            }
+            return false;
+        }
+    }
+
+    private async getSDKs(existingPath : string) : Promise<IDotnetListInfo[]>
+    {
+        const findSDKsCommand = CommandExecutor.makeCommand(existingPath, ['--list-sdks']);
+
+        const sdkInfo = await (this.executor!).execute(findSDKsCommand).then((result) =>
+            {
+                const runtimes = result.stdout.split('\n').map((line) => line.trim()).filter((line) => line.length > 0);
+                const runtimeInfos : IDotnetListInfo[] = runtimes.map((sdk) =>
+                {
+                    if(sdk === '') // new line in output that got trimmed
+                    {
+                        return null;
+                    }
+                    const parts = sdk.split(' ', 2); // account for spaces in PATH, no space should appear before then and luckily path is last
+                    return {
+                        mode: 'sdk',
+                        version: parts[0],
+                        directory: sdk.split(' ').slice(1).join(' ').slice(1, -1) // need to remove the brackets from the path [path]
+                    } as IDotnetListInfo;
+                }).filter(x => x !== null) as IDotnetListInfo[];
+
+                return runtimeInfos;
+            });
+
+            return sdkInfo;
+    }
+
+    private async getRuntimes(existingPath : string) : Promise<IDotnetListInfo[]>
+    {
+        const findRuntimesCommand = CommandExecutor.makeCommand(existingPath, ['--list-runtimes']);
+
+        const windowsDesktopString = 'Microsoft.WindowsDesktop.App';
+        const aspnetCoreString = 'Microsoft.AspNetCore.App';
+        const runtimeString = 'Microsoft.NETCore.App';
+
+        const runtimeInfo = await (this.executor!).execute(findRuntimesCommand).then((result) =>
+        {
+            const runtimes = result.stdout.split('\n').map((line) => line.trim()).filter((line) => line.length > 0);
+            const runtimeInfos : IDotnetListInfo[] = runtimes.map((runtime) =>
+            {
+                if(runtime === '') // new line in output that got trimmed
+                {
+                    return null;
+                }
+                const parts = runtime.split(' ', 3); // account for spaces in PATH, no space should appear before then and luckily path is last
+                return {
+                    mode: parts[0] === aspnetCoreString ? 'aspnetcore' : parts[0] === runtimeString ? 'runtime' : 'sdk', // sdk is a placeholder for windows desktop, will never match since this is for runtime search only
+                    version: parts[1],
+                    directory: runtime.split(' ').slice(2).join(' ').slice(1, -1) // account for spaces in PATH, no space should appear before then and luckily path is last.
+                    // the 2nd slice needs to remove the brackets from the path [path]
+                } as IDotnetListInfo;
+            }).filter(x => x !== null) as IDotnetListInfo[];
```

Comment:
nit: Same as above where it might be easier as RegEx. The logic looks similar too, so potential for deduplication.